src/app/credExplorer/weights/EdgeTypeConfig.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+import React from "react";
+import {WeightSlider, type Props as WeightSliderProps} from "./WeightSlider";
+import {type EdgeType} from "../../adapters/pluginAdapter";
+
+export type WeightedEdgeType = {|
+  +type: EdgeType,
+  +forwardWeight: number,
+  +backwardWeight: number,
+|};
+
+export function defaultWeightedEdgeType(type: EdgeType): WeightedEdgeType {
+  return {
+    type,
+    forwardWeight: 1,
+    backwardWeight: 1,
+  };
+}
+
+export class EdgeTypeConfig extends React.Component<{
+  +weightedType: WeightedEdgeType,
+  +onChange: (WeightedEdgeType) => void,
+}> {
+  render() {
+    // This component renders the "forward name" with the "backward weight" and
+    // vice versa. While counter-intuitive, this is the correct UI decision.
+    // This is because while edges are organized src->dst in the code, in the
+    // UI we always show dst->src, as the PagerankTable fixes a certain
+    // destination node, and then looks at the sources contributing to that
+    // node's score. If we don't reverse the name and weight here, the UI
+    // becomes very confusing.
+    //
+    // To see why, let us consider the case of REFERENCES edges. Suppose that
+    // the user opens the cred explorer and sees that the highest-scored pull
+    // request has most of its cred from REFERENCES connections, i.e. it got
+    // its cred from the references edges *backwards* components.  So for the
+    // user to decrease the cred from the REFERENCES connection, they actually
+    // must reduce the weight of the referenced_by backward weights.
+    //
+    // Intuitively, given that the user wants to reduce the REFERENCES
+    // connection, and they have an option to change weights on either
+    // REFERENCES or REFERENCED_BY, it is very intuitive that they should
+    // reduce the weight on REFERENCES. Hence, we swap the weights.
+    return (
+      <div>
+        <EdgeWeightSlider
+          name={this.props.weightedType.type.backwardName}
+          weight={this.props.weightedType.forwardWeight}
+          onChange={(forwardWeight) => {
+            this.props.onChange({
+              ...this.props.weightedType,
+              forwardWeight,
+            });
+          }}
+        />
+        <EdgeWeightSlider
+          name={this.props.weightedType.type.forwardName}
+          weight={this.props.weightedType.backwardWeight}
+          onChange={(backwardWeight) => {
+            this.props.onChange({
+              ...this.props.weightedType,
+              backwardWeight,
+            });
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+export function styledVariable(letter: string) {
+  return (
+    <span style={{fontWeight: 700, fontStyle: "italic", marginRight: 3}}>
```

Comment:
Please indicate that the `marginRight` is an _italic correction_.

Really, the unit of measurement should be `em`, not `px`, because the
dimension scales (linearly) with the font size. Something like `0.1em`
would probably be a good starting point.